bowbuffer.go

```diff
@@ -10,15 +10,20 @@ import (
 
 // Buffer is a mutable data structure with the purpose of easily building data Series with:
 // - Data: slice of data.
+// - DataType: type of the data.
 // - nullBitmapBytes: slice of bytes representing valid or null values.
 type Buffer struct {
 	Data            interface{}
+	DataType        Type
```

Comment:
Historically this was here. It have been a mistake to remove it in the first place since a given type can fit several arrow types.

Reply:
Indeed :)